### ISSUE TITLE
Contributing guidelines

### ISSUE THREAD

**Issue by Fernerkundung** [sentinelsat/sentinelsat#119]
I think it is time for us to discuss the creation of contributing guidelines. Benefit would be that it helps others who want to contribute to this project but also as a documentation for the rest of us. I don't think we need to go all out on this like [pandas](http://pandas.pydata.org/pandas-docs/stable/contributing.html) or [numpy](https://docs.scipy.org/doc/numpy-dev/dev/index.html) but rather keep it relatively short, like [rasterio](https://github.com/mapbox/rasterio/blob/master/CONTRIBUTING.rst).

What are things we should include? I'll start with:

- [x] design principles / code style guide
- [x] developing environment / How to create a PR for a fix or feature?
- [x] testing / writing tests (Travis and local)
- [x] documentation and changelog
- [x] release and packaging

**Comment by valgur** [comment]
Contributing guidelines would be nice to have, sure. We could create a GitHub wiki page for it initially, so it's easier to collaborate on writing it, and if there's anything that needs discussing we could do it here or on Gitter.

**Comment by j08lue** [comment]
I made a first draft based on Rasterio's guidelines. It is not meant to be complete at all yet, just a starting point: https://github.com/sentinelsat/sentinelsat/wiki/CONTRIBUTING---draft

**Comment by Fernerkundung** [comment]
@j08lue I built upon your initial draft and tracked the changes in a [public gist: CONTRIBUTE.md](https://gist.github.com/Fernerkundung/20c7eb237a16164a111d7a25fe75984b).

Any further comments by @valgur or @willemarcel, especially regarding the design principles and code conventions?

**Comment by willemarcel** [comment]
It's good for me!

**Comment by valgur** [comment]
Looks very good! I appreciate the fact that you managed to condense our goals to just a couple of bullet points. 😮

Some thoughts:
- The wording of "sentinelsat is meant to be not more and not less than a Pythonic client..." sounds slightly off, although I agree with the idea. Perhaps "nothing more and nothing less"?
- I would add that all unit tests must be able to run without interacting with the server by default, e.g. by using vcrpy cassettes. This is pretty much enforced anyway, tough, since I removed SciHub credentials from TravisCI.

**Comment by j08lue** [comment]
> "nothing more and nothing less"

That is actually what I meant. Feel free to choose a less ambiguous formulation, too, if you like.

> all unit tests must be able to run without interacting with the server by default

Yes.

**Comment by Fernerkundung** [comment]
Or just?
>sentinelsat is a Pythonic client...

>all unit tests must be able to run without interacting with the server by default

New unit tests have to run against the Open Access Hub at least once to record the response, so I'd rephrase it maybe to:

> all unit tests must use prerecorded responses to Copernicus Open Access Hub. We use [VCR.py](https://github.com/kevin1024/vcrpy)

**Comment by Fernerkundung** [comment]
Included with PR #127
